### PR TITLE
Fix bugs with highlighting and searcher

### PR DESCRIPTION
**Pull request by edwinzhng**
* Fixed some bugs from very rare errors

### CODE REVIEW

**Comment by rodrigonogueira4** [comment]
Just curious, what were the errors?

Also, should we change the `try/except` to catch only this tokenization error? I'm afraid it might silently fail for other errors

**Comment by edwinzhng** [comment]
> Just curious, what were the errors?
> 
> Also, should we change the `try/except` to catch only this tokenization error? I'm afraid it might silently fail for other errors

1. The error with the searcher is if the query has a not in the latin encoding like any letters with accents, caused Pyserini to throw an error occasionally

2. For highlighting, because we cut off the max number of paragraphs to highlight, for the very last article, the first paragraph may have been highlighted but not the second one. Then we end up with the length of highlights == 1 which does not match the length of paragraphs

And yeah I agree I will add a specific exception

**Comment by lintool** [comment]
We had similar/related issues in Pyserini - check out: https://github.com/castorini/pyserini/issues?q=encode

tl;dr - is to use `text.encode('utf-8')` explicitly.

**Comment by edwinzhng** [comment]
> We had similar/related issues in Pyserini - check out: https://github.com/castorini/pyserini/issues?q=encode
> 
> tl;dr - is to use `text.encode('utf-8')` explicitly.

Should I make a PR in Pyserini? The searcher doesn't encode the query

https://github.com/castorini/pyserini/blob/master/pyserini/search/pysearch.py#L78

**Comment by lintool** [comment]
Ah, yes, that's an oversight on our part. Yes please. Can you also include a test case that makes sure this is fixed?
